tests/main/remodel-gadget/task.yaml

```diff
@@ -0,0 +1,124 @@
+summary: Test a remodel that switches to a new gadget
+environment:
+    OLD_GADGET: pc
+    NEW_GADGET: test-snapd-pc
+
+systems: [ubuntu-core-18-64]
+
+prepare: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    #shellcheck source=tests/lib/systemd.sh
+    . "$TESTSLIB"/systemd.sh
+    systemctl stop snapd.service snapd.socket
+    rm -rf /var/lib/snapd/assertions/*
+    rm -rf /var/lib/snapd/device
+    rm -rf /var/lib/snapd/state.json
+    mv /var/lib/snapd/seed/assertions/model model.bak
+    cp "$TESTSLIB"/assertions/developer1.account /var/lib/snapd/seed/assertions
+    cp "$TESTSLIB"/assertions/developer1.account-key /var/lib/snapd/seed/assertions
+    cp "$TESTSLIB"/assertions/developer1-pc-18.model /var/lib/snapd/seed/assertions
+    cp "$TESTSLIB"/assertions/testrootorg-store.account-key /var/lib/snapd/seed/assertions
+    # kick first boot again
+    systemctl start snapd.service snapd.socket
+    while ! snap changes | grep -q "Done.*Initialize system state"; do sleep 1; done
+
+restore: |
+    if [ "$TRUST_TEST_KEYS" = "false" ]; then
+        echo "This test needs test keys to be trusted"
+        exit
+    fi
+    #shellcheck source=tests/lib/systemd.sh
+    . "$TESTSLIB"/systemd.sh
+    systemctl stop snapd.service snapd.socket
+    rm -rf /var/lib/snapd/assertions/*
+    rm -rf /var/lib/snapd/device
+    rm -rf /var/lib/snapd/state.json
+    rm -f /var/lib/snapd/seed/assertions/developer1.account
+    rm -f /var/lib/snapd/seed/assertions/developer1.account-key
+    rm -f /var/lib/snapd/seed/assertions/developer1-pc-18.model
+    rm -f /var/lib/snapd/seed/assertions/testrootorg-store.account-key
+    cp model.bak /var/lib/snapd/seed/assertions/model
+    rm -f ./*.bak
+    # kick first boot again
+    systemctl start snapd.service snapd.socket
+    # wait for first boot to be done
+    snap wait system seed.loaded
+    # cleanup the canary file
+    rm -f /boot/efi/canary.txt
+    while ! snap changes | grep -q "Done.*Initialize system state"; do sleep 1; done
```

Comment:
(nitpick) I guess nowdays we could use the retry-tool for this.

Reply:
With retry tool this would have to be:
```
retry-tool sh -c 'snap changes | grep -q "Done.*Initialize system state"
```
TBH I'm not sure which one I prefer

Reply:
Right, same remark about constraining this.

Reply:
Wrapped with retry tool too